File: 796.Rotate-String.go
// https://leetcode.com/problems/rotate-string/description/
//
// algorithms
// Easy (48.63%)
// Total Accepted:    26.6k
// Total Submissions: 54.7k
// beats 100.0% of golang submissions

package leetcode

func rotateString(A string, B string) bool {
	if A == B {
		return true
	}

	lenA, lenB := len(A), len(B)

	if lenA != lenB {
		return false
	}

	for idx, _ := range A {
		if A[idx:] == B[:lenA-idx] && A[:idx] == B[lenA-idx:] {
			return true
		}
	}

	return false
}

File: 796.Rotate-String.java
// https://leetcode.com/problems/rotate-string/
//
// algorithms
// Easy (49.79%)
// Total Accepted: 56,587
// Total Submissions: 114,343


class Solution {
    public boolean rotateString(String A, String B) {
        if (A.equals(B)) {
            return true;
        }
        int lenA = A.length(), lenB = B.length();
        if (lenA != lenB) {
            return false;
        }

        for (int i = 1; i < lenA; i++) {
            if (A.substring(i).equals(B.substring(0, lenA - i))
                    && A.substring(0, i).equals(B.substring(lenA - i, lenA))) {
                return true;
            }
        }

        return false;
    }
}